src/Stormpath/DataStore/DefaultResourceFactory.php

```diff
@@ -29,6 +30,19 @@ public function __construct(InternalDataStore $dataStore)
         $this->dataStore = $dataStore;
     }
 
+    public function instantiateChildClass($className, $childClassName, array $constructorArgs)
```

Comment:
I'm not sure why do we need this method. It looks like it's just making a validation (which shouldn't be necessary when providing the class name), but not really adding any value. I'm also not sure what do we need this validation for.